pkg/cmd/pulumi/util.go

```diff
@@ -862,10 +865,43 @@ func getRefreshOption(proj *workspace.Project, refresh string) (bool, error) {
 		return true, nil
 	}
 
-	// the default functionality right now is to always skip a refresh
+	// if the user has passed a plan file then the default behaviour is to refresh
+	if planFilePath != "" {
+		return true, nil
+	}
```

Comment:
I think it is quite surprising to couple `--plan` to `--refresh`.  These are orthogonal options, and users can certainly choose to use both together, but to have the former imply the latter doesn't feel right.

I'm fairly sure we will not want to do this - but I'm interested to understand the motivating scenario for this better.